[BUG] Specviz: Line analysis does not work on two subsets using API

<!-- This comments are hidden when you submit the issue,
so you do not need to remove them! -->

<!-- Please have a search on our GitHub repository to see if a similar
issue has already been posted.
If a similar issue is closed, have a quick look to see if you are satisfied
by the resolution.
If not please go ahead and open an issue! -->

<!-- This is a public repository, please scrub out sensitive information! -->

**Describe the bug**
A test in #1275 exposed a bug where if the continuum subset is set to subset 1 and spectral subset is set to subset 2, a traceback is returned.

<details>

```
        plugin.spectral_subset_selected = 'Subset 2'
>       plugin.continuum_subset_selected = 'Subset 1'

test_line_analysis.py:263: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Users/javerbukh/opt/anaconda3/envs/jdaviz-dev-2.5/lib/python3.8/site-packages/traitlets/traitlets.py:606: in __set__
    self.set(obj, value)
/Users/javerbukh/opt/anaconda3/envs/jdaviz-dev-2.5/lib/python3.8/site-packages/traitlets/traitlets.py:595: in set
    obj._notify_trait(self.name, old_value, new_value)
/Users/javerbukh/opt/anaconda3/envs/jdaviz-dev-2.5/lib/python3.8/site-packages/traitlets/traitlets.py:1219: in _notify_trait
    self.notify_change(Bunch(
/Users/javerbukh/opt/anaconda3/envs/jdaviz-dev-2.5/lib/python3.8/site-packages/ipywidgets/widgets/widget.py:686: in notify_change
    super(Widget, self).notify_change(change)
/Users/javerbukh/opt/anaconda3/envs/jdaviz-dev-2.5/lib/python3.8/site-packages/traitlets/traitlets.py:1229: in notify_change
    return self._notify_observers(change)
/Users/javerbukh/opt/anaconda3/envs/jdaviz-dev-2.5/lib/python3.8/site-packages/traitlets/traitlets.py:1266: in _notify_observers
    c(event)
../line_analysis.py:275: in _calculate_statistics
    right_max = np.nanmax([np.nanmax(mark_x['right']), sr.upper.value])
<__array_function__ internals>:180: in nanmax
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = array([nan, nan]), axis = None, out = None, keepdims = <no value>, initial = <no value>, where = <no value>

    @array_function_dispatch(_nanmax_dispatcher)
    def nanmax(a, axis=None, out=None, keepdims=np._NoValue, initial=np._NoValue,
               where=np._NoValue):
        """
        Return the maximum of an array or maximum along an axis, ignoring any
        NaNs.  When all-NaN slices are encountered a ``RuntimeWarning`` is
        raised and NaN is returned for that slice.
    
        Parameters
        ----------
        a : array_like
            Array containing numbers whose maximum is desired. If `a` is not an
            array, a conversion is attempted.
        axis : {int, tuple of int, None}, optional
            Axis or axes along which the maximum is computed. The default is to compute
            the maximum of the flattened array.
        out : ndarray, optional
            Alternate output array in which to place the result.  The default
            is ``None``; if provided, it must have the same shape as the
            expected output, but the type will be cast if necessary. See
            :ref:`ufuncs-output-type` for more details.
    
            .. versionadded:: 1.8.0
        keepdims : bool, optional
            If this is set to True, the axes which are reduced are left
            in the result as dimensions with size one. With this option,
            the result will broadcast correctly against the original `a`.
    
            If the value is anything but the default, then
            `keepdims` will be passed through to the `max` method
            of sub-classes of `ndarray`.  If the sub-classes methods
            does not implement `keepdims` any exceptions will be raised.
    
            .. versionadded:: 1.8.0
        initial : scalar, optional
            The minimum value of an output element. Must be present to allow
            computation on empty slice. See `~numpy.ufunc.reduce` for details.
    
            .. versionadded:: 1.22.0
        where : array_like of bool, optional
            Elements to compare for the maximum. See `~numpy.ufunc.reduce`
            for details.
    
            .. versionadded:: 1.22.0
    
        Returns
        -------
        nanmax : ndarray
            An array with the same shape as `a`, with the specified axis removed.
            If `a` is a 0-d array, or if axis is None, an ndarray scalar is
            returned.  The same dtype as `a` is returned.
    
        See Also
        --------
        nanmin :
            The minimum value of an array along a given axis, ignoring any NaNs.
        amax :
            The maximum value of an array along a given axis, propagating any NaNs.
        fmax :
            Element-wise maximum of two arrays, ignoring any NaNs.
        maximum :
            Element-wise maximum of two arrays, propagating any NaNs.
        isnan :
            Shows which elements are Not a Number (NaN).
        isfinite:
            Shows which elements are neither NaN nor infinity.
    
        amin, fmin, minimum
    
        Notes
        -----
        NumPy uses the IEEE Standard for Binary Floating-Point for Arithmetic
        (IEEE 754). This means that Not a Number is not equivalent to infinity.
        Positive infinity is treated as a very large number and negative
        infinity is treated as a very small (i.e. negative) number.
    
        If the input has a integer type the function is equivalent to np.max.
    
        Examples
        --------
        >>> a = np.array([[1, 2], [3, np.nan]])
        >>> np.nanmax(a)
        3.0
        >>> np.nanmax(a, axis=0)
        array([3.,  2.])
        >>> np.nanmax(a, axis=1)
        array([2.,  3.])
    
        When positive infinity and negative infinity are present:
    
        >>> np.nanmax([1, 2, np.nan, np.NINF])
        2.0
        >>> np.nanmax([1, 2, np.nan, np.inf])
        inf
    
        """
        kwargs = {}
        if keepdims is not np._NoValue:
            kwargs['keepdims'] = keepdims
        if initial is not np._NoValue:
            kwargs['initial'] = initial
        if where is not np._NoValue:
            kwargs['where'] = where
    
        if type(a) is np.ndarray and a.dtype != np.object_:
            # Fast, but not safe for subclasses of ndarray, or object arrays,
            # which do not implement isnan (gh-9009), or fmax correctly (gh-8975)
            res = np.fmax.reduce(a, axis=axis, out=out, **kwargs)
            if np.isnan(res).any():
>               warnings.warn("All-NaN slice encountered", RuntimeWarning,
                              stacklevel=3)
E               RuntimeWarning: All-NaN slice encountered


```

</details>

**To Reproduce**
Steps to reproduce the behavior:
1. Go to `test_line_analysis`
2. Uncomment the line after the `#TODO` statement
3. Run `pytest -s .` from within the line analysis tests directory
4. See error

--- a/jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py
+++ b/jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py
@@ -1,5 +1,7 @@
-import numpy as np
 import os
+import warnings
+
+import numpy as np
 from glue.core.message import (SubsetDeleteMessage,
                                SubsetUpdateMessage)
 from glue_jupyter.common.toolbar_vuetify import read_icon
@@ -266,10 +268,15 @@ class LineAnalysis(PluginTemplateMixin, DatasetSelectMixin, SpectralSubsetSelect
             else:
                 mark_x = {'left': spectral_axis_nanmasked[spectral_axis.value < sr.lower.value],
                           'right': spectral_axis_nanmasked[spectral_axis.value > sr.upper.value]}
-                # center should extend (at least) across the line region between the full
-                # range defined by the continuum subset(s)
-                left_min = np.nanmin([np.nanmin(mark_x['left']), sr.lower.value])
-                right_max = np.nanmax([np.nanmax(mark_x['right']), sr.upper.value])
+                # Center should extend (at least) across the line region between the full
+                # range defined by the continuum subset(s).
+                # OK for mark_x to be all NaNs.
+                with warnings.catch_warnings():
+                    warnings.simplefilter('ignore', category=RuntimeWarning)
+                    mark_x_min = np.nanmin(mark_x['left'])
+                    mark_x_max = np.nanmax(mark_x['right'])
+                left_min = np.nanmin([mark_x_min, sr.lower.value])
+                right_max = np.nanmax([mark_x_max, sr.upper.value])
                 mark_x['center'] = np.array([left_min, right_max])
 
         continuum_x = spectral_axis[continuum_mask].value

--- a/jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py
+++ b/jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py
@@ -205,11 +205,8 @@ class LineAnalysis(PluginTemplateMixin, DatasetSelectMixin, SpectralSubsetSelect
             self.update_results(None)
             return
 
-        if self.spectral_subset_selected != "Surrounding":
-            sr = self.app.get_subsets_from_viewer("spectrum-viewer",
-                                                  subset_type="spectral").get(self.spectral_subset_selected) # noqa
-        else:
-            sr = None
+        sr = self.app.get_subsets_from_viewer("spectrum-viewer",
+                                              subset_type="spectral").get(self.spectral_subset_selected) # noqa
 
         if self.spectral_subset_selected == "Entire Spectrum":
             spectrum = full_spectrum

--- a/jdaviz/configs/specviz/plugins/line_analysis/tests/test_line_analysis.py
+++ b/jdaviz/configs/specviz/plugins/line_analysis/tests/test_line_analysis.py
@@ -256,17 +256,14 @@ def test_continuum_subset_spectral_subset2(specviz_helper, spectrum1d):
     specviz_helper.app.state.drawer = True
 
     plugin = specviz_helper.app.get_tray_item_from_name('specviz-line-analysis')
-    assert 'Subset 1' in plugin.spectral_subset.labels
-    assert 'Subset 2' in plugin.spectral_subset.labels
+    assert plugin.spectral_subset.labels == ['Entire Spectrum', 'Subset 1', 'Subset 2']
 
-    # TODO: Find why uncommenting the following line
-    #  causes a ValueError: zero-size array
-    # plugin.spectral_subset_selected = 'Subset 2'
+    plugin.spectral_subset_selected = 'Subset 2'
     plugin.continuum_subset_selected = 'Subset 1'
     plugin.width = 3
 
     # Values have not yet been validated
-    assert np.allclose(float(plugin.results[0]['result']), -264.7195451356822, atol=1e-5)
+    np.testing.assert_allclose(float(plugin.results[0]['result']), 32.520521, atol=1e-5)
 
 
 def test_continuum_surrounding_no_right(specviz_helper, spectrum1d):

--- a/jdaviz/configs/specviz/plugins/line_analysis/tests/test_line_analysis.py
+++ b/jdaviz/configs/specviz/plugins/line_analysis/tests/test_line_analysis.py
@@ -2,6 +2,7 @@ import astropy.units as u
 from astropy.table import QTable
 import numpy as np
 from glue.core.roi import XRangeROI
+from glue.core.edit_subset_mode import NewMode
 from ipywidgets.widgets import widget_serialization
 
 from jdaviz.configs.specviz.plugins.line_analysis.line_analysis import _coerce_unit
@@ -62,6 +63,9 @@ def test_line_identify(specviz_helper, spectrum1d):
     # will default to no selection
     assert la_plugin.selected_line == ''
 
+    # check redshift
+    assert la_plugin.selected_line_redshift == 0.0
+
     # but selecting a line from line-list (or clicking) should change the dropdown value
     # since sync is enabled by default
     assert la_plugin.sync_identify is True
@@ -82,6 +86,23 @@ def test_line_identify(specviz_helper, spectrum1d):
     assert ll_plugin.list_contents['Test List']['lines'][0].get('identify') is True
     assert ll_plugin.list_contents['Test List']['lines'][1].get('identify') is False
 
+    # selected_line should become the same as identified_line
+    assert la_plugin.selected_line == "Halpha 6563.0"
+    assert la_plugin.identified_line == "O III 5007.0"
+    la_plugin.sync_identify = True
+    assert la_plugin.selected_line == "O III 5007.0"
+    assert la_plugin.identified_line == "O III 5007.0"
+
+    # identified_line should become the same as selected_line
+    la_plugin.identified_line = ''
+    la_plugin.sync_identify = False
+    la_plugin.sync_identify = True
+    assert la_plugin.identified_line == "O III 5007.0"
+
+    # manually update redshift
+    la_plugin.vue_line_assign()
+    assert la_plugin.selected_line_redshift == -1.0
+
 
 def test_coerce_unit():
     q_input = 1 * u.Unit('1E-20 erg m / (Angstrom cm**2 s)')
@@ -94,3 +115,248 @@ def test_coerce_unit():
     q_input.uncertainty = None
     q_coerced = _coerce_unit(q_input)
     assert not hasattr(q_coerced, 'uncertainty')
+
+
+def test_continuum_surrounding_spectral_subset(specviz_helper, spectrum1d):
+    label = "Test 1D Spectrum"
+    specviz_helper.load_spectrum(spectrum1d, data_label=label)
+
+    specviz_helper.app.state.drawer = True
+    tray_names = [ti['name'] for ti in specviz_helper.app.state.tray_items]
+    plugin_index = tray_names.index('specviz-line-analysis')
+    specviz_helper.app.state.tray_items_open = [plugin_index]
+
+    # continuum should be created, plotted, and visible
+    sv = specviz_helper.app.get_viewer('spectrum-viewer')
+    continuum_marks = [m for m in sv.figure.marks if isinstance(m, LineAnalysisContinuum)]
+    assert len(continuum_marks) == 3
+    assert np.all([cm.visible for cm in continuum_marks])
+
+    # add a region and rerun stats for that region
+    sv.apply_roi(XRangeROI(6500, 7400))
+    specviz_helper.app.state.drawer = True
+
+    plugin = specviz_helper.app.get_tray_item_from_name('specviz-line-analysis')
+    assert 'Subset 1' in plugin.spectral_subset.labels
+    plugin.continuum_subset_selected = 'Surrounding'
+    plugin.spectral_subset_selected = 'Subset 1'
+    plugin.width = 3
+
+    # Values have not yet been validated
+    assert np.allclose(float(plugin.results[0]['result']), 350.89288537581467, atol=1e-5)
+
+
+def test_continuum_spectral_same_value(specviz_helper, spectrum1d):
+    label = "Test 1D Spectrum"
+    specviz_helper.load_spectrum(spectrum1d, data_label=label)
+
+    specviz_helper.app.state.drawer = True
+    tray_names = [ti['name'] for ti in specviz_helper.app.state.tray_items]
+    plugin_index = tray_names.index('specviz-line-analysis')
+    specviz_helper.app.state.tray_items_open = [plugin_index]
+
+    # continuum should be created, plotted, and visible
+    sv = specviz_helper.app.get_viewer('spectrum-viewer')
+    continuum_marks = [m for m in sv.figure.marks if isinstance(m, LineAnalysisContinuum)]
+    assert len(continuum_marks) == 3
+    assert np.all([cm.visible for cm in continuum_marks])
+
+    # add a region and rerun stats for that region
+    sv.apply_roi(XRangeROI(6500, 7400))
+    specviz_helper.app.state.drawer = True
+
+    plugin = specviz_helper.app.get_tray_item_from_name('specviz-line-analysis')
+    assert 'Subset 1' in plugin.spectral_subset.labels
+    plugin.continuum_subset_selected = 'Subset 1'
+    plugin.spectral_subset_selected = 'Subset 1'
+    plugin.width = 3
+
+    # continuum and spectral cannot be the same value
+    assert plugin.results[0]['result'] == ''
+
+
+def test_continuum_surrounding_invalid_width(specviz_helper, spectrum1d):
+    label = "Test 1D Spectrum"
+    specviz_helper.load_spectrum(spectrum1d, data_label=label)
+
+    specviz_helper.app.state.drawer = True
+    tray_names = [ti['name'] for ti in specviz_helper.app.state.tray_items]
+    plugin_index = tray_names.index('specviz-line-analysis')
+    specviz_helper.app.state.tray_items_open = [plugin_index]
+
+    # continuum should be created, plotted, and visible
+    sv = specviz_helper.app.get_viewer('spectrum-viewer')
+    continuum_marks = [m for m in sv.figure.marks if isinstance(m, LineAnalysisContinuum)]
+    assert len(continuum_marks) == 3
+    assert np.all([cm.visible for cm in continuum_marks])
+
+    # add a region and rerun stats for that region
+    sv.apply_roi(XRangeROI(6500, 7400))
+    specviz_helper.app.state.drawer = True
+
+    plugin = specviz_helper.app.get_tray_item_from_name('specviz-line-analysis')
+    assert 'Subset 1' in plugin.spectral_subset.labels
+    plugin.continuum_subset_selected = 'Surrounding'
+    plugin.spectral_subset_selected = 'Subset 1'
+    plugin.width = 11
+    assert plugin.results[0]['result'] == ''
+
+
+def test_continuum_subset_spectral_entire(specviz_helper, spectrum1d):
+    label = "Test 1D Spectrum"
+    specviz_helper.load_spectrum(spectrum1d, data_label=label)
+
+    specviz_helper.app.state.drawer = True
+    tray_names = [ti['name'] for ti in specviz_helper.app.state.tray_items]
+    plugin_index = tray_names.index('specviz-line-analysis')
+    specviz_helper.app.state.tray_items_open = [plugin_index]
+
+    # continuum should be created, plotted, and visible
+    sv = specviz_helper.app.get_viewer('spectrum-viewer')
+    continuum_marks = [m for m in sv.figure.marks if isinstance(m, LineAnalysisContinuum)]
+    assert len(continuum_marks) == 3
+    assert np.all([cm.visible for cm in continuum_marks])
+
+    # add a region and rerun stats for that region
+    sv.apply_roi(XRangeROI(6500, 7400))
+    specviz_helper.app.state.drawer = True
+
+    plugin = specviz_helper.app.get_tray_item_from_name('specviz-line-analysis')
+    assert 'Subset 1' in plugin.spectral_subset.labels
+    plugin.continuum_subset_selected = 'Subset 1'
+    plugin.spectral_subset_selected = 'Entire Spectrum'
+    plugin.width = 3
+
+    # Values have not yet been validated
+    assert np.allclose(float(plugin.results[0]['result']), -467.6854635447396, atol=1e-5)
+
+
+def test_continuum_subset_spectral_subset2(specviz_helper, spectrum1d):
+    label = "Test 1D Spectrum"
+    specviz_helper.load_spectrum(spectrum1d, data_label=label)
+
+    specviz_helper.app.state.drawer = True
+    tray_names = [ti['name'] for ti in specviz_helper.app.state.tray_items]
+    plugin_index = tray_names.index('specviz-line-analysis')
+    specviz_helper.app.state.tray_items_open = [plugin_index]
+
+    # continuum should be created, plotted, and visible
+    sv = specviz_helper.app.get_viewer('spectrum-viewer')
+    continuum_marks = [m for m in sv.figure.marks if isinstance(m, LineAnalysisContinuum)]
+    assert len(continuum_marks) == 3
+    assert np.all([cm.visible for cm in continuum_marks])
+
+    # add a region and rerun stats for that region
+    sv.apply_roi(XRangeROI(6200, 7000))
+    specviz_helper.app.state.drawer = True
+
+    sv.session.edit_subset_mode._mode = NewMode
+    sv.session.edit_subset = []
+    sv.apply_roi(XRangeROI(7100, 7700))
+    specviz_helper.app.state.drawer = True
+
+    plugin = specviz_helper.app.get_tray_item_from_name('specviz-line-analysis')
+    assert 'Subset 1' in plugin.spectral_subset.labels
+    assert 'Subset 2' in plugin.spectral_subset.labels
+
+    # TODO: Find why uncommenting the following line
+    #  causes a ValueError: zero-size array
+    # plugin.spectral_subset_selected = 'Subset 2'
+    plugin.continuum_subset_selected = 'Subset 1'
+    plugin.width = 3
+
+    # Values have not yet been validated
+    assert np.allclose(float(plugin.results[0]['result']), -264.7195451356822, atol=1e-5)
+
+
+def test_continuum_surrounding_no_right(specviz_helper, spectrum1d):
+    label = "Test 1D Spectrum"
+    specviz_helper.load_spectrum(spectrum1d, data_label=label)
+
+    specviz_helper.app.state.drawer = True
+    tray_names = [ti['name'] for ti in specviz_helper.app.state.tray_items]
+    plugin_index = tray_names.index('specviz-line-analysis')
+    specviz_helper.app.state.tray_items_open = [plugin_index]
+
+    # continuum should be created, plotted, and visible
+    sv = specviz_helper.app.get_viewer('spectrum-viewer')
+    continuum_marks = [m for m in sv.figure.marks if isinstance(m, LineAnalysisContinuum)]
+    assert len(continuum_marks) == 3
+    assert np.all([cm.visible for cm in continuum_marks])
+
+    # add a region and rerun stats for that region
+    sv.apply_roi(XRangeROI(6500, 8000))
+    specviz_helper.app.state.drawer = True
+
+    plugin = specviz_helper.app.get_tray_item_from_name('specviz-line-analysis')
+    assert 'Subset 1' in plugin.spectral_subset.labels
+
+    plugin.spectral_subset_selected = 'Subset 1'
+    plugin.continuum_subset_selected = 'Surrounding'
+    plugin.width = 3
+
+    # Values have not yet been validated
+    assert np.allclose(float(plugin.results[0]['result']), 39.76685499263615, atol=1e-5)
+
+
+def test_continuum_surrounding_no_left(specviz_helper, spectrum1d):
+    label = "Test 1D Spectrum"
+    specviz_helper.load_spectrum(spectrum1d, data_label=label)
+
+    specviz_helper.app.state.drawer = True
+    tray_names = [ti['name'] for ti in specviz_helper.app.state.tray_items]
+    plugin_index = tray_names.index('specviz-line-analysis')
+    specviz_helper.app.state.tray_items_open = [plugin_index]
+
+    # continuum should be created, plotted, and visible
+    sv = specviz_helper.app.get_viewer('spectrum-viewer')
+    continuum_marks = [m for m in sv.figure.marks if isinstance(m, LineAnalysisContinuum)]
+    assert len(continuum_marks) == 3
+    assert np.all([cm.visible for cm in continuum_marks])
+
+    # add a region and rerun stats for that region
+    sv.apply_roi(XRangeROI(6000, 7500))
+    specviz_helper.app.state.drawer = True
+
+    plugin = specviz_helper.app.get_tray_item_from_name('specviz-line-analysis')
+    assert 'Subset 1' in plugin.spectral_subset.labels
+
+    plugin.spectral_subset_selected = 'Subset 1'
+    plugin.continuum_subset_selected = 'Surrounding'
+    plugin.width = 3
+
+    # Values have not yet been validated
+    assert np.allclose(float(plugin.results[0]['result']), 146.67186446784513, atol=1e-5)
+
+
+def test_subset_changed(specviz_helper, spectrum1d):
+    label = "Test 1D Spectrum"
+    specviz_helper.load_spectrum(spectrum1d, data_label=label)
+
+    specviz_helper.app.state.drawer = True
+    tray_names = [ti['name'] for ti in specviz_helper.app.state.tray_items]
+    plugin_index = tray_names.index('specviz-line-analysis')
+    specviz_helper.app.state.tray_items_open = [plugin_index]
+
+    # continuum should be created, plotted, and visible
+    sv = specviz_helper.app.get_viewer('spectrum-viewer')
+    continuum_marks = [m for m in sv.figure.marks if isinstance(m, LineAnalysisContinuum)]
+    assert len(continuum_marks) == 3
+    assert np.all([cm.visible for cm in continuum_marks])
+
+    # add a region and rerun stats for that region
+    sv.apply_roi(XRangeROI(6000, 7500))
+    specviz_helper.app.state.drawer = True
+
+    plugin = specviz_helper.app.get_tray_item_from_name('specviz-line-analysis')
+    assert 'Subset 1' in plugin.spectral_subset.labels
+
+    plugin.spectral_subset_selected = 'Subset 1'
+    plugin.continuum_subset_selected = 'Surrounding'
+    plugin.width = 3
+
+    sv.apply_roi(XRangeROI(6500, 7500))
+    specviz_helper.app.state.drawer = True
+
+    # Values have not yet been validated
+    assert np.allclose(float(plugin.results[0]['result']), 350.89288537581467, atol=1e-5)

--- a/jdaviz/configs/specviz/plugins/line_analysis/tests/test_line_analysis.py
+++ b/jdaviz/configs/specviz/plugins/line_analysis/tests/test_line_analysis.py
@@ -143,7 +143,7 @@ def test_continuum_surrounding_spectral_subset(specviz_helper, spectrum1d):
     plugin.width = 3
 
     # Values have not yet been validated
-    assert np.allclose(float(plugin.results[0]['result']), 350.89288537581467, atol=1e-5)
+    np.testing.assert_allclose(float(plugin.results[0]['result']), 350.89288537581467, atol=1e-5)
 
 
 def test_continuum_spectral_same_value(specviz_helper, spectrum1d):
@@ -228,7 +228,7 @@ def test_continuum_subset_spectral_entire(specviz_helper, spectrum1d):
     plugin.width = 3
 
     # Values have not yet been validated
-    assert np.allclose(float(plugin.results[0]['result']), -467.6854635447396, atol=1e-5)
+    np.testing.assert_allclose(float(plugin.results[0]['result']), -467.6854635447396, atol=1e-5)
 
 
 def test_continuum_subset_spectral_subset2(specviz_helper, spectrum1d):
@@ -293,7 +293,7 @@ def test_continuum_surrounding_no_right(specviz_helper, spectrum1d):
     plugin.width = 3
 
     # Values have not yet been validated
-    assert np.allclose(float(plugin.results[0]['result']), 39.76685499263615, atol=1e-5)
+    np.testing.assert_allclose(float(plugin.results[0]['result']), 39.76685499263615, atol=1e-5)
 
 
 def test_continuum_surrounding_no_left(specviz_helper, spectrum1d):
@@ -323,7 +323,7 @@ def test_continuum_surrounding_no_left(specviz_helper, spectrum1d):
     plugin.width = 3
 
     # Values have not yet been validated
-    assert np.allclose(float(plugin.results[0]['result']), 146.67186446784513, atol=1e-5)
+    np.testing.assert_allclose(float(plugin.results[0]['result']), 146.67186446784513, atol=1e-5)
 
 
 def test_subset_changed(specviz_helper, spectrum1d):
@@ -356,4 +356,4 @@ def test_subset_changed(specviz_helper, spectrum1d):
     specviz_helper.app.state.drawer = True
 
     # Values have not yet been validated
-    assert np.allclose(float(plugin.results[0]['result']), 350.89288537581467, atol=1e-5)
+    np.testing.assert_allclose(float(plugin.results[0]['result']), 350.89288537581467, atol=1e-5)

